tests/test_layouts.py

```diff
@@ -250,6 +251,74 @@ def testMDS(self):
         lo = g.layout("mds")
         self.assertTrue(isinstance(lo, Layout))
 
+    def testUMAP(self):
+        g = Graph()
+
+        # These distinct types of exceptions look ugly... can we improve?
```

Comment:
At some point some of the catch-all igraph `InternalError` exceptions should be converted to proper Python exceptions, but that's API-breaking and I'm not sure whether there's a straightforward mapping that makes sense in most contexts. The only information we can use in our error handler to decide whether we want to convert to a Python exception is the igraph error code, and that may not always be enough.

Let's leave it to the subject of another issue / PR.